### PR TITLE
CAMEL-10466  - fix/dropbox oom

### PR DESCRIPTION
**Pull request by ecausarano**
See CAMEL-10466 - In a nutshell: I've changed the DropboxAPIFacade to use an OutputStreamBuilder rather than a BAOS. All other changes are corollary to this.

I'm still missing some questions about testing, returning empty content etc... (see discussion in JIRA)

### CODE REVIEW

**Comment by davsclaus** [comment]
Can you run

   mvn compile -Psourcecheck

In the camel-dropbox directory and fix the checkstyle errors

File: components/camel-dropbox/pom.xml [hunk 1]
```diff
@@ -107,6 +107,18 @@
                 <artifactId>camel-maven-plugin</artifactId>
                 <version>${project.version}</version>
             </plugin>
+            <plugin>
```
Comment (davsclaus) [inline components/camel-dropbox/pom.xml]: Why did you add this maven plugin

File: components/camel-dropbox/src/main/java/org/apache/camel/component/dropbox/core/DropboxAPIFacade.java [hunk 2]
```diff
@@ -257,58 +235,65 @@ public DropboxResult move(String remotePath, String newRemotePath) throws Dropbo
...
+    }
+
+
+    private Map<String, Object> downloadFilesInFolder(String path) throws DropboxException {
+        try {
+            DbxEntry.WithChildren listing = client.getMetadataWithChildren(path);
+            if (listing == null) {
+                return Collections.emptyMap();
+            } else if (listing.children == null) {
+                LOG.info("downloading a single file...");
```
Comment (davsclaus) [inline components/camel-dropbox/src/main/java/org/apache/camel/component/dropbox/core/DropboxAPIFacade.java]: This should maybe be DEBUG logging level to reduce logging noise

File: components/camel-dropbox/src/main/java/org/apache/camel/component/dropbox/core/DropboxAPIFacade.java [hunk 3]
```diff
@@ -257,58 +235,65 @@ public DropboxResult move(String remotePath, String newRemotePath) throws Dropbo
...
+                return Collections.singletonMap(entry.getKey(), entry.getValue());
+            }
+            Map<String, Object> result = new HashMap<>();
+            for (DbxEntry entry : listing.children) {
+                if (entry.isFile()) {
+                    try {
+                        Map.Entry<String, Object> singleFile = downloadSingleFile(entry.path);
+                        result.put(singleFile.getKey(), singleFile.getValue());
+                    } catch (DropboxException e) {
+                        LOG.warn("can't download from " + entry.path);
```
Comment (davsclaus) [inline components/camel-dropbox/src/main/java/org/apache/camel/component/dropbox/core/DropboxAPIFacade.java]: LOG.warn("Cannot download from " + entry.path + " due " + e.getMessage() + ". This exception is ignored.");

File: components/camel-dropbox/src/main/java/org/apache/camel/component/dropbox/core/DropboxAPIFacade.java [hunk 4]
```diff
@@ -257,58 +235,65 @@ public DropboxResult move(String remotePath, String newRemotePath) throws Dropbo
...
-    private void downloadSingleFile(String path, Map<String, ByteArrayOutputStream> resultEntries) throws DropboxException {
-        ByteArrayOutputStream baos = new ByteArrayOutputStream();
-        DbxEntry.File downloadedFile;
+    private Map.Entry<String, Object> downloadSingleFile(String path) throws DropboxException {
         try {
-            downloadedFile = DropboxAPIFacade.client.getFile(path, null, baos);
+            OutputStreamBuilder target = OutputStreamBuilder.withExchange(exchange);
+            DbxEntry.File downloadedFile = client.getFile(path, null, target);
+            if (downloadedFile != null) {
+                LOG.info("downloaded path:" + path);
```
Comment (davsclaus) [inline components/camel-dropbox/src/main/java/org/apache/camel/component/dropbox/core/DropboxAPIFacade.java]: Debug level

File: components/camel-dropbox/src/main/java/org/apache/camel/component/dropbox/dto/DropboxMoveResult.java [hunk 5]
```diff
@@ -16,21 +16,35 @@
...
+    }
+
+    public String getOldPath() {
+        return oldPath;
     }
+
+    public String getNewPath() {
+        return newPath;
+    }
+
```
Comment (davsclaus) [inline components/camel-dropbox/src/main/java/org/apache/camel/component/dropbox/dto/DropboxMoveResult.java]: Remove the code below that are not in use

File: components/camel-dropbox/src/main/java/org/apache/camel/component/dropbox/dto/DropboxResult.java [hunk 6]
```diff
@@ -29,11 +31,11 @@
      */
     public abstract void populateExchange(Exchange exchange);
 
-    public Object getResultEntries()  {
-        return resultEntries;
-    }
+//    public Map<String, OutputStream> getResultEntries()  {
```
Comment (davsclaus) [inline components/camel-dropbox/src/main/java/org/apache/camel/component/dropbox/dto/DropboxResult.java]: Remove code that are not in use

File: components/camel-dropbox/pom.xml (hunk 1 above)
Comment (ecausarano) [inline components/camel-dropbox/pom.xml]: Without it "mvn install" from within the component module doesn't create the sources jar (perhaps I'm doing it wrong?)
